Fix dead link to NumPy docs in random functions (#2384)
cupy/cupy: docs/source/reference/random.rst

Random Sampling (``cupy.random``)
=================================

.. https://docs.scipy.org/doc/numpy-1.16.1/reference/routines.random.html
.. https://docs.scipy.org/doc/numpy/reference/random/legacy.html

Differences between :mod:`cupy.random` and :mod:`numpy.random`:

> * CuPy currently provides `Legacy Random Generation <https://docs.scipy.org/doc/numpy/reference/random/legacy.html>`_ API (see also: `NumPy 1.16 Reference <https://docs.scipy.org/doc/numpy-1.16.1/reference/routines.random.html>`_).

Reviewer: ```suggestion
* CuPy provides `Legacy Random Generation <https://docs.scipy.org/doc/numpy/reference/random/legacy.html>`_ API (see also: `NumPy 1.16 Reference <https://docs.scipy.org/doc/numpy-1.16.1/reference/routines.random.html>`_).
```

>   New random generator API (:class:`numpy.random.Generator` class) introduced in NumPy 1.17 is not implemented yet.

Reviewer: ```suggestion
  The new random generator API (:class:`numpy.random.Generator` class) introduced in NumPy 1.17 has not been implemented yet.
```

> * Most functions under :mod:`cupy.random` supports ``dtype`` option, which does not exist in the corresponding NumPy APIs.

Reviewer: ```suggestion
* Most functions under :mod:`cupy.random` support the ``dtype`` option, which do not exist in the corresponding NumPy APIs.
```

>   This option enables us to generate float32 values directly without any space overhead.

Reviewer: ```suggestion
  This option enables generation of float32 values directly without any space overhead.
```

* CuPy does not guarantee that the same number generator is used across major versions.

>   This means that numbers generated by :mod:`cupy.random` of a new major version may not be the same as that of its previous one even if the same seed and distribution are used.

Reviewer: ```suggestion
  This means that numbers generated by :mod:`cupy.random` by new major version may not be the same as the previous one, even if the same seed and distribution are used.
```
